eras/conway/impl/src/Cardano/Ledger/Conway/Rules/Epoch.hs

```diff
@@ -172,7 +198,7 @@ returnProposalDeposits removedProposals oldUMap =
                 (RewDepUView um)
             , unclaimed
             )
-        | otherwise = (um, unclaimed <> gasDeposit gas)
+        | otherwise = (um, Map.insertWith (<>) (gasId gas) (gasDeposit gas) unclaimed)
```

Comment:
So instead of adding the  values inside the function, we're returning them as a map so we have the values "indexed" by the id, in order to use this info in the Event - is my understanding correct?

Reply:
Yes, exactly. We need the IDs so we can say which proposals are unclaimed in the `GovInfoEvent`.